### PR TITLE
3214 cannot read property slice of undefined

### PR DESCRIPTION
**Pull request by omerzam**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

About # ([3214](https://app.zenhub.com/workspaces/gooddollar-5d50833888a83c6880d3e345/issues/gooddollar/gooddapp/3214))

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: src/server/verification/verificationAPI.js [hunk 1]
```diff
@@ -491,7 +491,7 @@ const setup = (app: Router, verifier: VerificationAPI, gunPublic: StorageAPI, st
       const { user, body } = req
       const verificationData: { code: string } = body.verificationData
       const { email } = user.otp || {}
-      const hashedNewEmail = sha3(email)
+      const hashedNewEmail = email ? sha3(email) : null
```
Comment (sirpy) [inline src/server/verification/verificationAPI.js:494]: what is the reason that we get empty email from client side? has it been addressed?

Comment (omerzam) [reply src/server/verification/verificationAPI.js:494]: from the client-side we get the otp code
the email we get from the user object

Comment (sirpy) [reply src/server/verification/verificationAPI.js:494]: ok sorry, still the question is why is the data is malformed in the first place? why is the user trying to verify an email that we dont have?

**Comment by omerzam** [comment]
the root cause of this issue should be resolved by https://github.com/GoodDollar/GoodDAPP/pull/3235

**Comment by serdiukov-o-nordwhale** [comment]
@omerzam it was merged. could we close this PR ? or you would merge it with master and put on review again ?

**Comment by omerzam** [comment]
> @omerzam #3235 was merged. could we close this PR ? or you would merge it with master and put on review again ?

@serdiukov-o-nordwhale  waiting on #3235 to pass QA and be released to make sure the number of such errors is reduced to nearly 0, then I will merge this PR

**Comment by sirpy** [comment]
@omerzam @serdiukov-o-nordwhale this is open for almost a month

**Pull request merged**